File: API/src/constants/ocupacaoAnimalConstants.ts
// src/constants/OcupacaoAnimalConstants.ts

export enum StatusOcupacaoAnimal {
    ATIVO = 1,
    REMOVIDO = 2
}

// Mapeamento dos códigos para descrições
export const StatusOcupacaoAnimalDescriptions: Record<StatusOcupacaoAnimal, string> = {
    [StatusOcupacaoAnimal.ATIVO]: 'Ativo',
    [StatusOcupacaoAnimal.REMOVIDO]: 'Removido',
};

File: mobile/lib/enums/ocupacao_animal_constants.dart
// lib/enums/ocupacao_animak_constants.dart

/// Enum representando o status da ocupacao do animal.
enum StatusOcupacaoAnimal {
  ativo,
  removido,
}

/// Descrições para o StatusOcupacaoAnimal.
const Map<StatusOcupacaoAnimal, String> statusOcupacaoAnimalDescriptions = {
  StatusOcupacaoAnimal.ativo: 'ativo',
  StatusOcupacaoAnimal.removido: 'removido'
};

/// Mapeamento de valores inteiros para StatusOcupacaoAnimal.
const Map<int, StatusOcupacaoAnimal> intToStatusOcupacaoAnimal = {
  1: StatusOcupacaoAnimal.ativo,
  2: StatusOcupacaoAnimal.removido,
};

/// Mapeamento de StatusOcupacaoAnimal para valores inteiros.
const Map<StatusOcupacaoAnimal, int> statusOcupacaoToInt = {
  StatusOcupacaoAnimal.ativo: 1,
  StatusOcupacaoAnimal.removido: 2,
};